packages/block-editor/src/components/spacing-sizes-control/spacing-input-control.js

```diff
@@ -0,0 +1,250 @@
+	};
+
+	const handleCustomValueSliderChange = ( next ) => {
+		onChange( [ next, selectedUnit ].join( '' ) );
+	};
+
+	const allPlaceholder = isMixed ? __( 'Mixed' ) : null;
+
+	const currentValueHint = ! isMixed
+		? customTooltipContent( currentValue )
+		: __( 'Mixed' );
+
+	const options = spacingSizes.map( ( size, index ) => ( {
+		key: index,
+		name: size.name,
+	} ) );
+
+	const marks = spacingSizes.map( ( newValue, index ) => ( {
+		value: index,
+		label: undefined,
+	} ) );
+
+	const ariaLabel = sprintf(
+		// translators: 1: The side of the block being modified (top, bottom, left, etc.). 2. Type of spacing being modified (Padding, margin, etc)
+		__( '%1$s %2$s' ),
+		LABELS[ side ],
+		type?.toLowerCase()
+	);
+
+	return (
+		<>
+			{ side !== 'all' && (
+				<HStack className="components-spacing-sizes-control__side-labels">
+					<Text className="components-spacing-sizes-control__side-label">
+						{ LABELS[ side ] }
+					</Text>
+
+					{ spacingSizes.length <= 9 &&
+						! showCustomValueControl &&
+						currentValueHint !== undefined && (
+							<Text className="components-spacing-sizes-control__hint-single">
+								{ currentValueHint }
+							</Text>
+						) }
+				</HStack>
+			) }
+			{ side === 'all' &&
+				spacingSizes.length <= 9 &&
+				! showCustomValueControl &&
+				currentValueHint !== undefined && (
+					<Text className="components-spacing-sizes-control__hint-all">
+						{ currentValueHint }
+					</Text>
+				) }
+
+			<Button
+				label={
+					showCustomValueControl
+						? __( 'Use size preset' )
+						: __( 'Set custom size' )
+				}
+				icon={ settings }
+				onClick={ () => {
+					setShowCustomValueControl( ! showCustomValueControl );
+				} }
+				isPressed={ showCustomValueControl }
+				isSmall
+				className={ classnames( {
+					'components-spacing-sizes-control__custom-toggle-all':
+						side === 'all',
+					'components-spacing-sizes-control__custom-toggle-single':
+						side !== 'all',
+				} ) }
+				iconSize={ 24 }
+			/>
+			{ showCustomValueControl && (
+				<>
+					<UnitControl
+						onChange={ ( newSize ) =>
+							onChange( getNewCustomValue( newSize ) )
+						}
+						value={ currentValue }
+						units={ units }
+						placeholder={ allPlaceholder }
+						disableUnits={ isMixed }
+						label={ ariaLabel }
+						hideLabelFromVision={ true }
+						className="components-spacing-sizes-control__custom-value-input"
+					/>
+
+					<RangeControl
+						value={ customRangeValue }
+						min={ 0 }
+						max={ 100 }
+						withInputField={ false }
+						onChange={ handleCustomValueSliderChange }
+						className="components-spacing-sizes-control__custom-value-range"
+					/>
+				</>
+			) }
+			{ spacingSizes.length <= 9 && ! showCustomValueControl && (
+				<RangeControl
+					className={ classnames(
+						'components-spacing-sizes-control__range-control',
+						{
+							'component-spacing-sizes-control__not-set':
+								value === undefined,
+						}
+					) }
```

Comment:
```suggestion
					className="components-spacing-sizes-control__range-control"
```

I didn't see any uses of the `component-spacing-sizes-control__not-set` class. Can we remove it?

Reply:
a hangover from previous iterations - gone.